File: gen_seed.js

const fs = require('fs');
const path = require('path');

const filePath = 'c:/Users/NathanSR/Projects/gymaux-app/src/config/seedExercises.ts';
const content = fs.readFileSync(filePath, 'utf-8');

// Function to clean up values
const clean = (val) => val ? val.trim().replace(/^['"]|['"]$/g, '') : null;

// More robust extraction
// We want to skip commented out lines
const lines = content.split('\n');
const cleanLines = lines.filter(line => !line.trim().startsWith('//'));
const cleanContent = cleanLines.join('\n');

const objectRegex = /\{([\s\S]+?)\}/g;
let match;
let count = 0;
let sql = "-- Seed exercises\n";

while ((match = objectRegex.exec(cleanContent)) !== null) {
    const block = match[1];
    
    // Extract fields within the block
    const idMatch = block.match(/id:\s*(\d+)/);
    const nameMatch = block.match(/name:\s*["']([^"']+)["']/);
    const descMatch = block.match(/description:\s*["']([^"']+)["']/);
    const catMatch = block.match(/category:\s*["']([^"']+)["']/);
    const tagsMatch = block.match(/tags:\s*\[([\s\S]+?)\]/);
    const howToMatch = block.match(/howTo:\s*["']([^"']+)["']/);

    if (!idMatch || !nameMatch) continue;

    const id = idMatch[1];
    const name = nameMatch[1];
    const description = descMatch ? descMatch[1] : '';
    const category = catMatch ? catMatch[1] : '';
    const tagsRaw = tagsMatch ? tagsMatch[1] : '';
    const howTo = howToMatch ? howToMatch[1] : '';

    const tags = tagsRaw.split(',').map(t => clean(t)).filter(t => t);
    const tagsSql = tags.length > 0 ? "ARRAY[" + tags.map(t => `'${t}'`).join(', ') + "]" : "ARRAY[]::text[]";

    sql += `INSERT INTO exercises (id, name, description, category, tags, how_to, created_by_type, is_public, created_by) VALUES (${id}, '${name}', '${description}', '${category}', ${tagsSql}, '${howTo}', 'system', true, null) ON CONFLICT (id) DO NOTHING;\n`;
    count++;
}

fs.writeFileSync('c:/Users/NathanSR/Projects/gymaux-app/seed_exercises.sql', sql);
console.log(`Generated SQL with ${count} exercises.`);

File: gen_seed.py

import re
import json

def parse_exercises(content):
    # This regex is a bit naive but should work for this specific file structure
    # It looks for blocks like { id: ..., name: ..., ... }
    exercise_blocks = re.findall(r'\{\s+id: (\d+),\s+name: "([^"]+)",\s+description: "([^"]+)",\s+category: "([^"]+)",\s+tags: \[([^\]]+)\],\s+howTo: "([^"]+)"\s+\}', content)
    
    exercises = []
    for block in exercise_blocks:
        tags = [t.strip().strip('"').strip("'") for t in block[4].split(',')]
        exercises.append({
            "id": int(block[0]),
            "name": block[1],
            "description": block[2],
            "category": block[3],
            "tags": tags,
            "how_to": block[5],
            "created_by_type": 'system',
            "is_public": True,
            "created_by": None
        })
    return exercises

with open('c:/Users/NathanSR/Projects/gymaux-app/src/config/seedExercises.ts', 'r', encoding='utf-8') as f:
    content = f.read()

exercises = parse_exercises(content)

# Generate SQL
sql = "-- Seed exercises\n"
for ex in exercises:
    tags_sql = "ARRAY[" + ", ".join([f"'{t}'" for t in ex['tags']]) + "]"
    sql += f"INSERT INTO exercises (id, name, description, category, tags, how_to, created_by_type, is_public, created_by) VALUES ({ex['id']}, '{ex['name']}', '{ex['description']}', '{ex['category']}', {tags_sql}, '{ex['how_to']}', 'system', true, null) ON CONFLICT (id) DO NOTHING;\n"

with open('c:/Users/NathanSR/Projects/gymaux-app/seed_exercises.sql', 'w', encoding='utf-8') as f:
    f.write(sql)

print(f"Generated SQL with {len(exercises)} exercises.")
